src/instance.rs

```diff
@@ -898,9 +898,17 @@ where
     ///     // This reference's lifetime is determined by `py`'s lifetime.
     ///     // Because that originates from outside this function,
     ///     // this return value is allowed.
+    ///     # #[allow(deprecated)]
     ///     obj.into_ref(py)
     /// }
     /// ```
+    #[cfg_attr(
+        not(feature = "gil-refs"),
+        deprecated(
+            since = "0.21.0",
+            note = "`Py::into_ref` will be replaced by `Py::into_bound` in a future PyO3 version"
```

Comment:
I think here it would be slightly better to use the same wording we went for in https://github.com/PyO3/pyo3/pull/3856/files#diff-b36e5dab91de9ae329fe692f3b0db5aec5fbd720042069742b401c7169cc184eR833

(This is because the `is_ref` name is completely deprecated and will be removed, we're not going to overwrite it later, and I picked the word "replaced" to imply the function itself will change.)